### ISSUE TITLE
Add GraalPy support

### ISSUE THREAD

**Issue by timfel** [actions/setup-python#693]
**Description:**
I would like to add support for installing [GraalPy](https://github.com/oracle/graalpython) via setup-python. We are starting to ask people to include support for GraalPy in their packages, and they (reasonably) request to be able to test against GraalPy using their existing workflows using setup-python.

**Are you willing to submit a PR?**
I have prepared the changes for setup-python, mimicking the PyPy support code. The tests using local mock data pass.

**Comment by timfel** [comment]
> Hello @timfel. Thank you for your feature request. We'll investigate it.

Thanks. I opened a PR to implement this at https://github.com/actions/setup-python/pull/694

**Comment by dmitry-shibanov** [comment]
Hello @timfel. We've merged the pull request. Could you please try to use `actions/setup-python@main` and confirm that it works as expected for you. For now I'm going to reopen the issue until the new version of action is released.

**Comment by kurtmckee** [comment]
I tested this using the following configuration:

```yaml
- uses: "actions/setup-python@v5"
  with:
    python-version: |
      graalpy-23.1
      pypy-3.10
      3.11
```

[Link to run, which failed on Windows](https://github.com/kurtmckee/detect-pythons/actions/runs/7169923447/job/19521432415?pr=12#step:2:14)

For Linux and macOS, `graalpy-23.1` resolved to version [23.1.1](https://github.com/oracle/graalpython/releases/tag/graal-23.1.1).

However, on Windows, [`setup-python` reported that no "23.1" version existed](https://github.com/kurtmckee/detect-pythons/actions/runs/7169923447/job/19521432415?pr=12#step:2:14):

```
Run actions/setup-python@v5
  with:
    python-version: graalpy-23.1
  pypy-3.10
  3.11
  
    check-latest: false
    token: ***
    update-environment: true
    allow-prereleases: false
Installed versions
  GraalPy version 23.1 was not found in the local cache
  Error: GraalPy version 23.1 with arch x64 not found
```

This is the complete list of files for the GraalPy 23.1.1 release at the time of writing:

```
graalpy-23.1.1-linux-aarch64.tar.gz
graalpy-23.1.1-linux-aarch64.tar.gz.sha256
graalpy-23.1.1-linux-amd64.tar.gz
graalpy-23.1.1-linux-amd64.tar.gz.sha256
graalpy-23.1.1-macos-aarch64.tar.gz
graalpy-23.1.1-macos-aarch64.tar.gz.sha256
graalpy-23.1.1-macos-amd64.tar.gz
graalpy-23.1.1-macos-amd64.tar.gz.sha256
graalpy-23.1.1-windows-amd64.zip
graalpy-23.1.1-windows-amd64.zip.sha256
graalpy-community-23.1.1-linux-aarch64.tar.gz
graalpy-community-23.1.1-linux-aarch64.tar.gz.sha256
graalpy-community-23.1.1-linux-amd64.tar.gz
graalpy-community-23.1.1-linux-amd64.tar.gz.sha256
graalpy-community-23.1.1-macos-aarch64.tar.gz
graalpy-community-23.1.1-macos-aarch64.tar.gz.sha256
graalpy-community-23.1.1-macos-amd64.tar.gz
graalpy-community-23.1.1-macos-amd64.tar.gz.sha256
graalpy-community-jvm-23.1.1-linux-aarch64.tar.gz
graalpy-community-jvm-23.1.1-linux-aarch64.tar.gz.sha256
graalpy-community-jvm-23.1.1-linux-amd64.tar.gz
graalpy-community-jvm-23.1.1-linux-amd64.tar.gz.sha256
graalpy-community-jvm-23.1.1-macos-aarch64.tar.gz
graalpy-community-jvm-23.1.1-macos-aarch64.tar.gz.sha256
graalpy-community-jvm-23.1.1-macos-amd64.tar.gz
graalpy-community-jvm-23.1.1-macos-amd64.tar.gz.sha256
graalpy-jvm-23.1.1-linux-aarch64.tar.gz
graalpy-jvm-23.1.1-linux-aarch64.tar.gz.sha256
graalpy-jvm-23.1.1-linux-amd64.tar.gz
graalpy-jvm-23.1.1-linux-amd64.tar.gz.sha256
graalpy-jvm-23.1.1-macos-aarch64.tar.gz
graalpy-jvm-23.1.1-macos-aarch64.tar.gz.sha256
graalpy-jvm-23.1.1-macos-amd64.tar.gz
graalpy-jvm-23.1.1-macos-amd64.tar.gz.sha256
graalpy-jvm-23.1.1-windows-amd64.zip
graalpy-jvm-23.1.1-windows-amd64.zip.sha256
```

It appears that there is an x64-compatible Windows release available, and I expected this to result in GraalPy getting installed for Windows.

Is this perhaps an issue with Windows releases using `.zip` as their file extension?

**Comment by timfel** [comment]
@kurtmckee The windows support of graalpy 23.1 is highly experimental, so we excluded it on purpose. The 24 release due out April 24 will improve the situation at which point we'll update the action

**Comment by kurtmckee** [comment]
Okay, thanks for the quick response!

**Comment by dmitry-shibanov** [comment]
Hello everyone. For now I'm going to close the issue because the support for GraalPy was released in [4.8.0](https://github.com/actions/setup-python/releases/tag/v4.8.0).
